crates/ty_python_semantic/src/types/signatures.rs

```diff
@@ -1963,8 +1905,24 @@ impl<'db> Parameters<'db> {
         }
     }
 
-    /// Return parameters that represents `(*args: object, **kwargs: object)`.
-    pub(crate) fn object() -> Self {
+    /// Return the "top" parameters (infinite union of all possible parameters).
+    pub(crate) fn top() -> Self {
+        Self {
+            // We always emit `called-top-callable` for any call to the top callable (based on the
+            // `kind` below), so we otherwise give it the most permissive signature`(*object,
+            // **object)`, so that we avoid emitting any other errors about arity mismatches.
+            value: vec![
+                Parameter::variadic(Name::new_static("args")).with_annotated_type(Type::object()),
+                Parameter::keyword_variadic(Name::new_static("kwargs"))
+                    .with_annotated_type(Type::object()),
+            ],
+            kind: ParametersKind::Top,
+        }
+    }
```

Comment:
It's very tempting, looking at this, to think "Ah, but surely the top-materialization of a `Callable` parameters can also be simplified, to `(*args: Never, **kwargs: Never)`, the inverse of the bottom-materialization of a `Callable` parameters directly below? But no, that's not the case, because those parameters would still permit calls that pass 0 arguments in -- it's actually "the same" as the empty parameters list `()`. It might be worth adding a comment here explicitly laying that out?